atox/src/main/kotlin/ui/user_profile/UserProfileFragment.kt

```diff
@@ -7,181 +7,205 @@ package ltd.evilcorp.atox.ui.user_profile
 import ltd.evilcorp.atox.R
 import ltd.evilcorp.atox.databinding.FragmentUserProfileBinding
+import ltd.evilcorp.atox.ui.AvatarMaker
 import ltd.evilcorp.atox.ui.BaseFragment
-import ltd.evilcorp.atox.ui.StatusDialog
+import ltd.evilcorp.atox.ui.colorFromStatus
+import ltd.evilcorp.atox.ui.dpToPx
+import ltd.evilcorp.atox.ui.isNightMode
+import ltd.evilcorp.atox.ui.setImageButtonRippleDayNight
 import ltd.evilcorp.atox.vmFactory
-import ltd.evilcorp.core.vo.UserStatus
-
-private const val TOX_MAX_NAME_LENGTH = 128
-private const val TOX_MAX_STATUS_MESSAGE_LENGTH = 1007
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import kotlin.math.min
+import kotlin.math.roundToInt
 
 private const val QR_CODE_TO_SCREEN_RATIO = 0.5f
-private const val QR_CODE_DIALOG_PADDING = 16f // in dp
-
-private fun dpToPx(dp: Float, res: Resources): Int =
-    TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, res.displayMetrics).toInt()
+private const val QR_CODE_PADDING = 16f // in dp
+private const val QR_CODE_SHARED_IMAGE_PADDING = 30f // in dp
 
 class UserProfileFragment : BaseFragment<FragmentUserProfileBinding>(FragmentUserProfileBinding::inflate) {
     private val vm: UserProfileViewModel by viewModels { vmFactory }
-    private lateinit var currentStatus: UserStatus
-
-    private fun colorFromStatus(status: UserStatus) = when (status) {
-        UserStatus.None -> ResourcesCompat.getColor(resources, R.color.statusAvailable, null)
-        UserStatus.Away -> ResourcesCompat.getColor(resources, R.color.statusAway, null)
-        UserStatus.Busy -> ResourcesCompat.getColor(resources, R.color.statusBusy, null)
-    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) = binding.run {
         ViewCompat.setOnApplyWindowInsetsListener(view) { _, compat ->
             val insets = compat.getInsets(WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.ime())
-            profileCollapsingToolbar.updatePadding(left = insets.left, right = insets.right)
-            profileToolbar.updatePadding(top = insets.top)
+            toolbar.updatePadding(left = insets.left, top = insets.top)
             mainSection.updatePadding(left = insets.left, right = insets.right)
             compat
         }
 
-        profileToolbar.apply {
-            setNavigationOnClickListener {
-                activity?.onBackPressed()
-            }
-        }
-
         vm.user.observe(viewLifecycleOwner) { user ->
-            currentStatus = user.status
-
             userName.text = user.name
             userStatusMessage.text = user.statusMessage
-            userStatus.setColorFilter(colorFromStatus(user.status))
+            profileImageLayout.statusIndicator.setColorFilter(colorFromStatus(resources, user.status))
+            AvatarMaker(user).setAvatar(profileImageLayout.profileImage)
         }
 
-        userToxId.text = vm.toxId.string()
+        // Inflating views according to Day/Night theme
```

Comment:
Isn't this usually handled by having different themes? It doesn't feel right that every single fragment needs to have custom code to handle light/dark mode since there's no code like that right now. :x
E.g. instead of having `ic_back_white` and `ic_back_black`, you have `ic_back` and set the tint of it using the theme?

Reply:
That's what I wanted but I didn't find a set tint method. I will look further if there's a different approach for that.

Reply:
I was doing something similar in https://github.com/evilcorpltd/aTox/blob/9ae23612034dd70d8a0e845d0cccac81cf37f874/atox/src/main/kotlin/ui/user_profile/UserProfileFragment.kt#L80
Maybe that helps?

Reply:
I'm trying with themes (https://coderedirect.com/questions/275631/tint-navigation-icon-in-toolbar)